### PR TITLE
Cy/server default exceptions

### PR DESCRIPTION
**Pull request by cy6erGn0m**
**Subsystem**
ktor-server-core

**Motivation**
When an http request contains an illegal Content-Type or Accept header value, ktor is producing a `BadContentTypeFormatException` that is reported as 500 and logged as regular exceptions. So log may contain a lot of such errors that is too noisy.

**Solution**
The better behaviour is to produce Bad Request (400) response and log error with debug level. This is much more convenient.

### CODE REVIEW

File: ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/DefaultEnginePipeline.kt [hunk 1]
```diff
@@ -101,6 +101,9 @@ private fun ApplicationEnvironment.logFailure(call: ApplicationCall, cause: Thro
             is CancellationException -> log.info("$status: $logString, cancelled")
             is ClosedChannelException -> log.info("$status: $logString, channel closed")
             is ChannelIOException -> log.info("$status: $logString, channel failed")
+            is BadRequestException -> log.debug("$status: $logString", cause)
+            is NotFoundException -> log.debug("$status: $logString", cause)
+            is UnsupportedMediaTypeException -> log.debug("$status: $logString, cause)")
```
Comment (e5l) [inline ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/DefaultEnginePipeline.kt]: Could you check the `"` correctness?

File: ktor-server/ktor-server-core/jvm/src/io/ktor/routing/RouteSelector.kt [hunk 2]
```diff
@@ -380,14 +381,19 @@ data class HttpHeaderRouteSelector(val name: String, val value: String) : RouteS
...
+            val parsedHeaders = parseAndSortContentTypeHeader(acceptHeaderContent)
+
+            if (parsedHeaders.isEmpty())
+                return RouteSelectorEvaluation.Missing
+            val header = parsedHeaders.firstOrNull { contentType.match(it.value) }
+            if (header != null)
+                return RouteSelectorEvaluation(true, header.quality)
+            return RouteSelectorEvaluation.Failed
+        } catch (failedToParse: BadContentTypeFormatException) {
+            throw BadRequestException("Illegal Accept header format: $acceptHeaderContent", failedToParse)
```
Comment (e5l) [inline ktor-server/ktor-server-core/jvm/src/io/ktor/routing/RouteSelector.kt]: Should we set the cause here?

File: ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/DefaultEnginePipeline.kt (hunk 1 above)
Comment (cy6erGn0m) [reply ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/DefaultEnginePipeline.kt]: fixed, thanks

**Comment by cy6erGn0m** [comment]
Merged

**Pull request closed without merging**